Mathlib/RingTheory/DedekindDomain/SInteger.lean

```diff
@@ -75,6 +75,35 @@ theorem integer_valuation_le_one (x : S.integer K) {v : HeightOneSpectrum R} (hv
     v.valuation K x ≤ 1 :=
   x.property v hv
 
+/-- If `S` is the whole set of places of `K`, then the `S`-integers are the whole of `K`. -/
+@[simp] lemma set_integer_top : (univ : Set (HeightOneSpectrum R)).integer K = ⊤ := by
+  ext
+  tauto
+
+/--
+If `S` is the empty set, then the `S`-integers are the minimal `R`-subalgebra of `K` (which is
+just `R` itself).
+-/
+@[simp]
+lemma set_integer_empty : (∅ : Set (HeightOneSpectrum R)).integer K = ⊥ := by
+  ext x
+  simp only [integer, mem_empty_iff_false, not_false_eq_true, true_implies]
+  refine ⟨IsDedekindDomain.HeightOneSpectrum.mem_integers_of_valuation_le_one K x, ?_⟩
+  rintro ⟨y, rfl⟩ v
+  exact v.valuation_le_one y
+
+variable (R) in
+/-- Canonical `R`-algebra isomorphism from `R` to the `∅`-integers of `K`. -/
+noncomputable def setIntegerEmptyEquiv : R ≃ₐ[R] (∅ : Set (HeightOneSpectrum R)).integer K :=
+  .ofBijective ((Algebra.ofId R K : R →ₐ[R] K).codRestrict _
+    (fun r ↦ by simpa only [set_integer_empty] using ⟨r, rfl⟩))
+    ⟨fun x y ↦ by simp [← SetLike.coe_eq_coe, Algebra.ofId_apply],
+      fun ⟨x, hx⟩ ↦ by simpa [← SetLike.coe_eq_coe, set_integer_empty, Algebra.mem_bot] using hx⟩
```

Comment:
Can you provide `Function.Bijective (algebraMap R ((∅ : Set (HeightOneSpectrum R)).integer K))` as a separate lemma? Thanks.

Reply:
I golfed the proof using the two things you mentioned, but also happy to remove it entirely. (I didn't find `Algebra.botEquivOfInjective` in the manual, I was searching for `Subalgebra` + `bot`.)

I also moved things that don't have a `Set` argument out of the `Set` namespace, and put them in `IsDedekindDomain` as well. (We seem to have `IsDedekindDomain` and `DedekindDomain` as separate namespaces, which is presumably an accident...)

Reply:
I would be inclined to remove it unless this equiv shows up often in your application.

Reply:
Off with its head, then...